python/cudf/cudf/core/column/string.py

```diff
@@ -199,15 +231,13 @@ def cat(self, others=None, sep=None, na_rep=None, **kwargs):
             If `others` is None, `str` is returned, otherwise a `Series/Index`
             (same type as caller) of str dtype is returned.
         """
-        from cudf.core import Series, Index
+        from cudf.core import Series, Index, DataFrame
```

Comment:
Would like to take a pass at simplifying this function in general. This is crazy complex compared to the other string APIs where if we could clean it up it would help for maintainability.